### ISSUE TITLE
Results different from paper

### ISSUE THREAD

**Issue by erfannoury** [ivendrov/order-embedding#3]
Hello,

As I was experimenting with the code and trying to reproduce the results reported in the paper, I noticed that they are different from each other. 
These are the values reported in the paper and values obtained from running the code (as is).

**Caption Retrieval**

| Method | R@1 | R@10 | Med r | Mean r |
| --- | --- | --- | --- | --- |
| Order-embedding (paper) | 46.7 | 88.9 | 2.0 | 5.7 |
| Order-embedding (code) | 47.2 | 90.7 | 2.0 | 4.7 |

**Image Retrieval**

| Method | R@1 | R@10 | Med r | Mean r |
| --- | --- | --- | --- | --- |
| Order-embedding (paper) | 37.9 | 85.9 | 2.0 | 8.1 |
| Order-embedding (code) | 39.5 | 87.5 | 2.0 | 7.1 |

Maybe the problem is with how I interpret the output of the code. Could you elaborate more on this?

**Comment by ivendrov** [comment]
Hi Erfan, thanks for reporting this! There will be some random variation in how well newly trained models do, but your results seem substantially better than the numbers I was seeing.

Could you tell me the exact sequence of commands you ran to get these numbers?

**Comment by erfannoury** [comment]
I just cloned the repository, downloaded all of the files listed in the readme and then used this command to train a model.

```
python driver.py order
```

**Comment by ivendrov** [comment]
The numbers it prints out during training are evaluated on the development set - admittedly this isn't well documented. To evaluate the trained model on the test set you then need to run

```
model = tools.load_model('snapshots/order')
evaluation.ranking_eval_5fold(model, split='test')
```

Since I tuned all the hyperparams on the development set, the performance on it should be a fair bit better than on the test set. But let me know if you observe otherwise!

**Comment by erfannoury** [comment]
Oh you're right. What a mistake!
I ran the evaluation code on the test split and these are the results. They are about the same as the ones reported in the paper. 

**Caption Retrieval**

| Method | R@1 | R@10 | Med r | Mean r |
| --- | --- | --- | --- | --- |
| Order-embedding (paper) | 46.7 | 88.9 | 2.0 | 5.7 |
| Order-embedding (code, test split) | 46.1 | 89.0 | 2.0 | 5.4 |

**Image Retrieval**

| Method | R@1 | R@10 | Med r | Mean r |
| --- | --- | --- | --- | --- |
| Order-embedding (paper) | 37.9 | 85.9 | 2.0 | 8.1 |
| Order-embedding (code, test split) | 37.8 | 86.1 | 2.0 | 7.5 |

Thank you Ivan for the correction. I think you can close this issue if you want.

**Comment by ivendrov** [comment]
Updated the training code to make it clearer where the numbers are coming from:  https://github.com/ivendrov/order-embedding/commit/2057e6056a887ea00d304fc43a8a7e4810f4d669
